### ISSUE TITLE
MultiThreadedAugmenter not abort_event.is_set()

### ISSUE THREAD

**Issue by lvlua123** [MIC-DKFZ/nnUNet#1892]
Hi,
Thanks for your awesome program.

I'm not using docker, but running the job through SLURM. The process hangs indefinitely at
2024-01-12 20:47:29.517981: lr: 0.01
using pin_memory on device 0
using pin_memory on device 0

Killing the process seems to indicate that it hangs within the MultiThreadedAugmenter. When I debug in PyCharm, I find that the program  loops under the if not abort_event.is_set(): statement of def producer(). I don't know how to fix it, please help!
![QQ截图20240112215739](https://github.com/MIC-DKFZ/nnUNet/assets/156439810/6c2670f8-d9d1-41d6-9069-41929f2d44d7)

Thanks!

**Comment by FabianIsensee** [comment]
Hey, 
this is a really hard one to solve as we don't have a slurm environment here to test things with. BUT I had access to a slurm cluster not too long ago and there everything worked perfectly.
Have you verified that other GPU/pytorch things run properly on your system? Does the GPU allocation work as it should? Do you have enough CPU cores allocated (>12)?
Best,
Fabian

**Comment by FabianIsensee** [comment]
Hey, what's the status? Have you been able to resolve this issue?

**Comment by lvlua123** [comment]
Hi, 
Thank you for your answer, but I still can't run the program properly.The GPU carries on taking up memory, but doesn't run (as shown in the red box), the program seems to hibernate.I don't know how to fix it, please help!
![Uploading QQ截图20240227104210.png…]()

Thanks！

**Comment by lvlua123** [comment]
Hi, 
Thank you for your answer, but I still can't run the program properly.The GPU carries on taking up memory, but doesn't run (as shown in the red box), the program seems to hibernate.I don't know how to fix it, please help!
![QQ截图20240227104210](https://github.com/MIC-DKFZ/nnUNet/assets/156439810/ddde63df-42ae-413e-8ee8-f579eb47461b)

Thanks！

**Comment by FabianIsensee** [comment]
OK So when the program loops in the producer (as indicated above) then this means that it is waiting for background workers to produce data for the forward pass. Something is holding them up and it is unclear what this is on your system. Here are a few suggestions:
- make sure you don't have all your processes bound to the same CPU slot (cpu bind in slurm). You need at least 12 CPU cores available to your job for nnU-Net to run reasonably well
- you might not have enough bandwidth to allow the background workers to load the data
- maybe you are stuck during the 'unpacking dataset' part. Please send the entire text ouput nnU-Net produces

Best,
Fabian

**Comment by guosijia** [comment]
I have the same problem, (on my private ubuntu server, RTX4090 GPU, not running with slurm). The problem should appear when the dataloader, the dataloader for the training set can get the data, while the dataloader for the validation set gets stuck at the first iteration when getting the data, at this point the program hangs and the gpu keeps hogging, this might be related to multiprocessing. However I have not been able to find a way to make it work.

**Comment by FabianIsensee** [comment]
Its really difficult for me to investigate this problem because I cannot reproduce it. Can you try to provide a code snippet that generates this error reliably also on other machines?
Can you manually debug into run_training and query both dataloaders?
```
train_batch = next(self.dataloader_train)
val_batch = next(self.dataloader_val)
```

**Comment by FabianIsensee** [comment]
Seems related to #2041
